tools/cli.js

```diff
@@ -134,8 +134,29 @@ commandProcessor.event.help = () => (
   'event <interfaceName> <eventName> [ <arg> [ , ... ] ]'
 );
 
-commandProcessor.connect = (host, port, appName, callback) => {
-  state.client = jstp.tcp.createClient({ host, port, secure: true });
+commandProcessor.connect = (protocol, host, port, appName, callback) => {
+  let transport;
+  const options = {};
+
+  switch (protocol) {
+    case 'jstps':
+      options.secure = true;
+      // fall through
+    case 'jstp':
+      transport = jstp.tcp;
+      options.host = host;
+      options.port = port;
+      break;
+    case 'ws':
+    case 'wss':
+      transport = jstp.ws;
+      options.url = `${protocol}://${host}:${port}`;
```

Comment:
Yeah, that is right, but we will still need to check for the transport type, which means that this `switch` statement won't be completely eliminated. Instead, it can be shortened a little bit.